server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ChunkOutputStream.java

```diff
@@ -0,0 +1,186 @@
+      log.trace("Establishing sink and listener!");
+      listener.set(new ChunkOutputStream.EventListener() {
+         @Override
+         public void data(final byte[] bs, final CompletableFuture<Boolean> demandedSignal) {
+             if (log.isTraceEnabled()) {
+                 log.trace("Sending {} bytes", bs.length);
+             }
+             sink.next(Tuples.of(bs, demandedSignal));
+         }
+         @Override
+         public void finish() {
+             sink.complete();
+         }
+      });
+      sinkCreated.complete(COMPLETED_SIGNAL);
+   });
+
+   /**
+    * The interface used to transfer data from an eventing API (i.e. {@link
+    * #asyncWrite}) to a {@link Flux} ({@link #out} in this case).  See
+    * documentation around {@link Flux#create} for more information on this.
+    */
+   interface EventListener {
+      void data(byte[] s, CompletableFuture<Boolean> demandedSignal);
+      void finish();
+   }
+
+   ChunkOutputStream(
+       final HttpServerResponse response,
+       final MonoProcessor<Void> completionMono
+   ) {
+      this.response = response;
+      this.completionMono = completionMono;
+   }
+
+   @Override
+   public void write(int b) {
+      listener.get().data(new byte[] {(byte)b}, new CompletableFuture<>());
+   }
+
+   @Override
+   public void close() throws IOException {
+       log.trace("Closing the ChunkOutputStream.");
+       final EventListener el = listener.get();
+       if (el != null) {
+           el.finish();
+       } else {
+           Mono.<Void>empty().subscribe(completionMono);
+       }
+       super.close();
+   }
+
+   @Override
+   public void write(byte[] bs, int off, int len) {
+       try {
+           asyncWrite(bs, off, len).get();
+       } catch (final InterruptedException ie) {
+           Thread.currentThread().interrupt();
+           throw new RuntimeException(ie);
+       } catch (final ExecutionException ee) {
+           throw new RuntimeException(ee);
+       }
+   }
+
+   @Override
+   public void flush() throws IOException {
+       log.trace("Flush called on ChunkOutputStream");
+      super.flush();
+   }
+
+   @Override
+   public CompletionStage<Void> asyncFlush() {
+      // TODO
+      return CompletableFuture.completedFuture(null);
+   }
+
+
+    @Override
+   public CompletableFuture<Void> asyncWrite(final byte[] bs, int offset, int length) {
+        if (!started) {
+            started = true;
+            Flux<byte[]> actualOut =
+                out.map(tuple -> {
+                    totalBytesSent += tuple.getT1().length;
+                    log.trace("{} total bytes sent", totalBytesSent);
+                    tuple.getT2().complete(COMPLETED_SIGNAL);
+                    return tuple.getT1();
+                }).doOnRequest(l -> log.trace("{} requested", l));
+            if (timeout != null) {
+                actualOut = actualOut.timeout(timeout);
+            }
+            response
+                .sendByteArray(actualOut)
```

Comment:
I just assumed that since we have `byte[]` that reactor-netty would do the best thing with `Flux<byte[]>`.  Seems weird that we'd do a better job converting it to a `ByteBuf`, but we can certainly look into it.